### PR TITLE
aabb: Move bounding box calculation into aabb.typ

### PR DESCRIPTION
**Pull request by johannes-wolf**
Simplify AABB calculation and move it to its own file.
Also support 3 dim vectors (for later use in some draw commands).

### CODE REVIEW

File: src/aabb.typ [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+/// - pts (array): List of points
+/// - init (dictionary): Initial aabb
+/// -> dictionary AABB object
+#let aabb(pts, init: none) = {
+  let bounds = init
+
+  if type(pts) == array {
+    for (i, pt) in pts.enumerate() {
+      if bounds == none and i == 0 {
+        bounds = (lo: pt, hi: pt)
```
Comment (fenjalien) [inline src/aabb.typ:17]: Not sure about `lo` and `hi`, maybe `min`/`max` or `low`/`high`?

File: src/aabb.typ [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        let (hi-x, hi-y, hi-z) = bounds.hi
+        bounds.hi = (calc.max(hi-x, x), calc.max(hi-y, y), calc.max(hi-z, z))
+      }
+    }
+    return bounds
+  } else if type(pts) == dictionary {
+    if init == none {
+      return pts
+    } else {
+      return aabb((pts.lo, pts.hi,), init: bounds)
```
Comment (fenjalien) [inline src/aabb.typ:36]: why not unpack the dictionary at the start of the funtion?

File: src/aabb.typ [hunk 3]
```diff
@@ -0,0 +1,61 @@
+/// Compute an axis aligned bounding box (aabb)
+/// for a list of points.
+///
+/// An aabb dictionary has the following keys:
+///  - lo (vector) Min. bounds vector
```
Comment (fenjalien) [inline src/aabb.typ:5]: Do we have a formal description of what a `vector` and a `point` type is?

File: src/aabb.typ [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+#let mid(bounds) = {
+  let (lo-x, lo-y, lo-z) = bounds.lo
+  let (hi-x, hi-y, hi-z) = bounds.hi
+  
+  return (lo-x + (hi-x - lo-x)/2,
+          lo-y + (hi-y - lo-y)/2,
+          lo-z + (hi-z - lo-z)/2)
+}
+
+/// Get the size of an aabb as vector
```
Comment (fenjalien) [inline src/aabb.typ:53]: can you be more specific that its the side lengths?

**Review by fenjalien — CHANGES_REQUESTED** [review]
its a good thing to move tbh. Although `aabb.aabb(...)` feels weird but its still a good name :)

File: src/aabb.typ (hunk 3 above)
Comment (johannes-wolf) [reply src/aabb.typ:5]: Not really. We use `vector` for arrays of numbers and `coordinate` for a cetz coordinate. Point should not be used as a type.

**Comment by johannes-wolf** [comment]
> its a good thing to move tbh. Although `aabb.aabb(...)` feels weird but its still a good name :)

Yeah, it feels weird. Could also name it `from-points(..)`?

File: src/aabb.typ (hunk 3 above)
Comment (fenjalien) [reply src/aabb.typ:5]: why not add one in `vector.typ`?

Comment (johannes-wolf) [reply src/aabb.typ:5]: > why not add one in `vector.typ`?

What do you mean?

Comment (fenjalien) [reply src/aabb.typ:5]: Add a description of the `vector` type there, given most should be created from there?

Comment (johannes-wolf) [reply src/aabb.typ:5]: We already use vectors everywhere but on the user API.
I actually think it is obvious that a vector is an array of scalars.

**Review by fenjalien — APPROVED** [review]

**Pull request merged**